### ISSUE TITLE
Salt-master does not process batch calls correctly after receiving too much data

### ISSUE THREAD

**Issue by keagan-thinkst** [saltstack/salt#55184]
### Description of Issue
Whenever I run a salt module function against my minions that returns too much data, I get the following error in the salt master's logs:

`[salt.transport.ipc:195 ][ERROR   ][28934] Exception occurred while handling stream: 11844764 exceeds max_str_len(1048576)`

Once this error has been seen, then any call from the master to a batch of minions fails on a random set of minions that seem to not reply back to the master. I test it with a simple `test.ping` with the `-ldebug` flag set.

Each time I run `test.ping` against * a different set of minions fail.

From the masters side the debug lines state that the master is not receiving any info from the minion, it retries the query once, then after that fails again it times out and then the call on the master finishes with output stating that the call failed against a random set of minions.

When I look into the salt logs of any of the minions that failed, I see that the minion is receiving all the correct data from the master, processing the request for the ping correctly and then connecting to the master with its response to the call. The master seemingly does not receive this communication from the minion and therefore the master retries. Again in the minions logs I see it receiving the retry attempt from the master, the minion processes the request and sends it onto the master successfully. But the master does not receive or process that data correctly and therefore the master sees the call as failing against the minion.
So to sum up, the minions are performing as expected but the master is not processing the info sent to it from a random set of minions correctly.

The master will continue to fail batch salt calls like this until I restart the salt master service. Then it runs again correctly until a module call returns too much data such as the log line posted above.

### Steps to Reproduce Issue
Run a module function that causes the minion to return too much data for the master, ie causing the following error in the master logs:
`[salt.transport.ipc:195 ][ERROR   ][28934] Exception occurred while handling stream: 11844764 exceeds max_str_len(1048576)`

Then performing any salt call from master against a large batch such as `test.ping` against *.

### Versions Report
Master:
```Salt Version:
           Salt: 2019.2.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.14
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.1
   mysql-python: 1.3.10
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15+ (default, Jul  9 2019, 16:51:35)
   python-gnupg: 0.4.1
         PyYAML: 5.1.2
          PyZMQ: 18.1.0
           RAET: Not Installed
          smmap: 2.0.5
        timelib: Not Installed
        Tornado: 5.1.1
            ZMQ: 4.3.2

System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-1032-aws
         system: Linux
        version: Ubuntu 18.04 bionic
```

Minions are the same except do not have the following installed:
```
dateutil
gitdb
gitpython
mysql-python
```

**Comment by frogunder** [comment]
@keagan-thinkst Thank you for reporting this issue.

Seems to be related to https://github.com/saltstack/salt/issues/55180

**Comment by arizvisa** [comment]
This bug is actually a duplicate of #54884 and is related to some outrageous defaults set by the version of msgpack being used. This bug is heavily discussed in #53230, so there's more info there too.

@keagan-thinkst, try fiddling around with the msgpack version as per these other issues. The msgpack module had some absurd constraints imposed in order to attempt to prevent denial-of-service attacks.

(Not trying to stop on your toes or anything, frogunder)

**Comment by keagan-thinkst** [comment]
Hi @arizvisa 

Thanks for your suggestion, however this does not seem to fix my problem. I pinned the msgpack version on my salt stack to 0.5.6 and still get the same behaviour where the salt master does not process the batch calls correctly after I run my module function. The module function returns a large dictionary back from the minions.
My current fix/work around has been to change the module function to only return very small dictionaries rather. Though am still very weary of the uncertainty in this case as I can reproduce this failing master state whenever a module function returns large amounts of information.

**Comment by arizvisa** [comment]
Ok then.. Well in that case, the only other option is literally to modify the calls to `msgpack.Unpacker` in `salt.transport.*` to add the `max_buffer_size` parameter set to a large value.

**Comment by keagan-thinkst** [comment]
Thank you very much @arizvisa !!!

I set `max_buffer_size=2147483647` in all calls to `msgpack.Unpacker` in `salt.transport.*` and I no longer get this issue! 

I also set my msgpack version back to `0.6.1` which is the suggested version for my salt deploy and the changes you suggested still fix my issue that I was seeing. 

Is this possibly going to be a configurable value to set in an upcoming version of salt so that the fix for the problem is easier to do than hardcoding/patching salt files?
